test/modularize_instance.cpp

```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
+#include <emscripten.h>
+#ifdef __EMSCRIPTEN_PTHREADS__
+#include <thread>
+#endif
+
+extern "C" EMSCRIPTEN_KEEPALIVE void foo() {
+  printf("foo\n");
+}
+
+extern "C" void bar() {
+  printf("bar\n");
+}
+
+int main() {
+  printf("main1\n");
+#ifdef __EMSCRIPTEN_PTHREADS__
+  std::thread([]{
+    printf("main2\n");
+  }).join();
```

Comment:
Can you using pthread_create here and keep this test as a single `C` test?   layering on C++ threading makes the test more complex than it needs to be.

Alternatively if you just want to test if threading works at all the easiest thing to do with run with `-sPROXY_TO_PTHREAD` .. then you don't need to write any of your own threading code.